memory/src/main/scala/filodb.memory/BlockManager.scala

```diff
@@ -221,8 +203,8 @@ class PageAlignedBlockManager(val totalMemorySizeInBytes: Long,
         val allocated = new Array[Block](num)
         (0 until num).foreach { i =>
           val block = freeBlocks.remove()
-          if (bmf.nonEmpty) block.setOwner(bmf.get)
-          use(block, bucketTime)
+          if (ownerBmf.nonEmpty) block.setOwner(ownerBmf.get)
+          use(block, odp)
```

Comment:
this block will be executed only if `odp = false`, otherwise `MemoryRequestException` gets thrown, right? and `use` method is not being used anywhere else.

Reply:
No - The foreach block is executed `num` number of times. It essentially takes the blocks from "free blocks list" and places them in the "used block list". It will be run for both ODP and ingestion.